odoo/tools/translate.py

```diff
@@ -443,8 +443,23 @@ def _get_lang(self, frame):
                     lang = env['res.users'].context_get()['lang']
         return lang
 
-    def __call__(self, source):
-        return self._get_translation(source)
+    def __call__(self, source, *args, **kw):
+        translation = self._get_translation(source)
+        if args:
+            try:
+                return translation % args
+            except (TypeError, ValueError):
+                bad = translation
+                translation = source % args
+                _logger.error('Bad translation "%r" for "%r"', bad, source)
+        elif kw:
```

Comment:
Could merge the two in getting the first non-emtpy of `args` and `kw`. Though we should probably also assert that we don't have *both* as that's nonsense.